interview-questions.md

```diff
@@ -8,52 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a block in Ruby?
 
   Your answer:
+I believe a block in Ruby is an anonymous function.
 
   Researched answer:
-
+A block in Ruby are anonymous functions that can be passed into methods. In plain terms, blocks are essentially, methods but have an anonymous piece of code, that can accept inputs in the form of an argument and return a value, but does not have a name. Blocks can be created with the do/end keywords or {}. Additionally, blocks can have multiple parameters.
 
 
 2. What is a gem?
 
   Your answer:
+I believe gem is a testing framework specifically for Ruby.
 
   Researched answer:
-
+Gem is a package manager specifically for ruby. Gem allows the basic formatting for storing and distributing programs and libraries to others. Rspec is an example of a package program that can be installed by using gem when using ruby.
 
 
 3. What is Ruby on Rails?
 
   Your answer:
+Ruby on Rails is a web application framework that allow us to interact with databases using ruby language.
 
   Researched answer:
-
+Ruby on rails is a server-side web application framework which is open-sourced. Rails uses general-purposes programing to build web apps, which means that it is the framework to build full-stack web applications. Additionally, rails uses MVC architectural pattern which stands for model, view, and controller.
 
 
 4. What is a relational database? Are there other kinds of databases?
 
   Your answer:
+A relational database is a type of database that stores data that are related to one another in a similar fashion to excel spreadsheet. There are many other types of databases.
 
   Researched answer:
-
+A relational database is a database that shares data by displaying it in rows and columns; similar to excel spreadsheet and google sheets. Schemas are a description of the layout of all the tables in the database.
 
 
 5. What are primary keys? Why are they important?
 
   Your answer:
+Primary keys are individual numbers that identify a certain value in the database. Each value/info in the database has a primary key to identify it.
 
   Researched answer:
-
+Primary keys are a column in a table that gives a value that uniquely identifies each row in a table.
```

Comment:
It will definitely serve you well to know the difference between primary and foreign keys.